DokanNet/BufferPool.cs

```diff
@@ -0,0 +1,173 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Threading;
+using DokanNet.Logging;
+
+namespace DokanNet
+{
+    /// <summary>
+    /// Simple buffer pool for buffers used by <see cref="IDokanOperations.ReadFile"/> and
+    /// <see cref="IDokanOperations.WriteFile"/> to avoid excessive Gen2 garbage collections due to large buffer
+    /// allocation on the large object heap (LOH).
+    /// 
+    /// This pool is a bit different than say System.Buffers.ArrayPool(T) in that <see cref="RentBuffer" /> only returns
+    /// exact size buffers. This is because the Read/WriteFile APIs only take a byte[] array as a parameter, not
+    /// buffer and length. As such, it would be back compat breaking to return buffers that are bigger than the
+    /// data length. To limit the amount of memory consumption, we only buffer sizes that are powers of 2, only
```

Comment:
Suggestion: expand how you arrived at powers of two, and that nothing really requires them to be powers of two.